symfony/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php

```diff
@@ -0,0 +1,212 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Authentication\Service;
+
+use OrangeHRM\Admin\Dto\UserSearchFilterParams;
+use OrangeHRM\Admin\Service\UserService;
+use OrangeHRM\Authentication\Dao\ResetPasswordDao;
+use OrangeHRM\Config\Config;
+use OrangeHRM\Core\Exception\ServiceException;
+use OrangeHRM\Core\Service\EmailService;
+use OrangeHRM\Core\Traits\ORM\EntityManagerHelperTrait;
+use OrangeHRM\Core\Traits\Service\DateTimeHelperTrait;
+use OrangeHRM\Core\Utility\Base64Url;
+use OrangeHRM\Entity\EmailConfiguration;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\Entity\ResetPassword;
+use OrangeHRM\Entity\User;
+
+class ResetPasswordService
+{
+    use DateTimeHelperTrait;
+    use EntityManagerHelperTrait;
+
+    public const RESET_PASSWORD_TOKEN_RANDOM_BYTES_LENGTH = 16;
+    protected ?EmailService $emailService = null;
+    protected ?userService $userService = null;
+    protected ?ResetPasswordDao $resetPasswordDao = null;
+
+    public function getEmailService(): ?EmailService
```

Comment:
```suggestion
    public function getEmailService(): EmailService
```
also missing doc comment